nsxt/resource_nsxt_policy_ipsec_vpn_service.go

```diff
@@ -203,11 +230,18 @@ func resourceNsxtPolicyIPSecVpnServiceRead(d *schema.ResourceData, m interface{}
 	if id == "" {
 		return fmt.Errorf("Error obtaining IPSecVpnService ID")
 	}
+	gatewayPath := d.Get("gateway_path").(string)
 	localeServicePath := d.Get("locale_service_path").(string)
+	if gatewayPath == "" && localeServicePath == "" {
+		return fmt.Errorf("At least one of gateway path and locale service path should be provided for VPN resources")
+	}
 	isT0, gwID, localeServiceID, err := parseLocaleServicePolicyPath(localeServicePath)
```

Comment:
Since this block is repeated below, can we move it to a function?

Reply:
done